app/state.go

```diff
@@ -656,7 +657,7 @@ func (u *updateCheckState) Handle(ctx *StateContext, c Controller) (State, bool)
 			// Just report successful update and return to normal operations.
 			return NewUpdateStatusReportState(update, client.StatusAlreadyInstalled), false
 		}
-		if err.Cause() == client.ErrNoDeploymentAvailable {
+		if errors.Cause(err) == client.ErrNoDeploymentAvailable {
```

Comment:
I've read that the best way to test whether an error is of a certain kind is to use [`errors.Is()`](https://golang.org/pkg/errors/#Is), since it will catch any error in the chain, not just top one (`err`) or bottom one (`errors.Cause(err)`).

Reply:
Nice! TIL